CONTRIBUTING.rst

```diff
@@ -0,0 +1,138 @@
+
+
+Ways to contribute
+------------------
+
+There are several options to contribute to this project:
+
+* Open a new topic on our `GitHub discussion <gh_discussions>`_ page.
+
+  Tell us about your ideas or ask questions there.
+  Discuss with us the next Towncrier release.
+
+* Help or comment on our GitHub `issues`_ tracker.
+
+  There are certainly many issues where you can help with your expertise.
+  Or you would like to share your ideas with us.
+
+* Open a new `issue <issues>`_.
+
+  If you found a bug or have a new cool feature, describe your findings.
+  Try to be as descriptive as possible to help us understand your issue.
+
+* :ref:`Modify the code <modify_code>`.
+
+  If you would love to see the new feature in the next release, this is
+  probably the best way.
+
+
+.. _modify_code:
+
+Modifying the Code
+------------------
+
+The source code is managed using Git and is hosted on GitHub::
+
+    https://github.com/twisted/towncrier
+    git@github.com:twisted/towncrier.git
+
+
+We recommend the following workflow:
+
+#. Fork our project on GitHub using this link:
+   https://github.com/twisted/towncrier/fork
+
+#. Clone your forked Git repository (replace ``GITHUB_USER`` with your
+   account name on GitHub)::
+
+   $ git clone git@github.com:GITHUB_USER/python-semver.git
+
+
+#. In your local repository create a pull request:
+
+   a. Create a new branch with::
+
+      $ git checkout -b <BRANCH_NAME>
+
+   b. Write your test cases and run the complete test suite, see :ref:`testsuite`
+      for details.
+
+   c. Document any user facing changes in the ``README.rst`` file.
+
+   d. Create a newsfragment in ``src/towncrier/newsfragments/``.
+
+   e. Create a `pull request`_.
+      Describe in the pull request what you did and why. If you have open questions, ask.
+
+#. Wait for feedback. If you receive any comments, address these.
+
+#. After your pull request got accepted, delete your branch.
+
+
+.. _testsuite:
+
+Running the Test Suite
+----------------------
+
+We use the default unittest module and `tox`_ to run tests against all supported
+Python versions and operation systems. All test dependencies are resolved
+automatically.
+
+You can decide to run the complete test suite or only part of it:
+
+* To run all tests, use::
+
+    $ tox
+
+  If you have not all Python interpreters installed on your system
+  it will probably give you some errors (``InterpreterNotFound``).
+  To avoid such errors, use::
+
+     $ tox --skip-missing-interpreters
+
+  It is possible to use one or more specific Python versions. Use the ``-e``
+  option and one or more abbreviations (``py36-tests`` for Python 3.6, ``py37-tests`` for
+  Python 3.7 etc.)::
+
+      $ tox -e py36-tests
+      $ tox -e py36-tests,py37-tests
+
+  To get a complete list and a short description, run::
+
+      $ tox -av
+
+* To run only a specific test only, use the ``towncrier.test.FILE.CLASS.METHOD`` syntax.
```

Comment:
for TDD I create a separate virtual env and call `trial` directy, include `trial -b towncrier` to drop me into a PDB session on errors so that I  can investigate what is wrong.

We can keep it. From my experience, tox is super slow of TDD.

Reply:
Ahh, now it gets interesting! :slightly_smiling_face:

I'm actually not really sure if we should remove that all and replace it with just a link to tox. This goes into kind of recommendations and how to develop things in towncrier.

Should we add that here or somewhere else?

I think, a general direction would be nice as this helps new contributors.

Reply:
OK, I've revised the "Running the Test Suite" section and shortened it in commit 436795a.